mithril.js

```diff
@@ -1488,18 +1488,26 @@
 		// m.route()
 		if (arguments.length === 0) return currentRoute
 		// m.route(el, defaultRoute, routes)
-		if (arguments.length === 3 && isString(arg1)) {
+		// m.route(el, defaultComponent, routes)
+		if (arguments.length === 3 && (isString(arg1) || isObject(arg1))) {
 			redirect = function (source) {
 				var path = currentRoute = normalizeRoute(source)
 				if (!routeByValue(root, arg2, path)) {
-					if (isDefaultRoute) {
-						throw new Error("Ensure the default route matches " +
-							"one of the routes defined in m.route")
+					// defaultRoute
+					if(isString(arg1)) {
```

Comment:
I should have moved `if(isDefaultRoute)` check inside `if(isString(arg1))` check?
I could provide another pull request.

```javascript
redirect = function (source) {
 				var path = currentRoute = normalizeRoute(source)
 				if (!routeByValue(root, arg2, path)) {
					// defaultRoute
					if(isString(arg1)) {
				        	if (isDefaultRoute) {
						      throw new Error("Ensure the default route matches " +
							"one of the routes defined in m.route")

```